ui/src/main/java/org/apache/hop/ui/hopgui/HopGui.java

```diff
@@ -1323,13 +1281,25 @@ protected void addPerspectivesToolbar() {
     formData.bottom = new FormAttachment(statusToolbar, 0);
     mainHopGuiComposite.setLayoutData(formData);
 
-    perspectivesToolbar = new ToolBar(mainHopGuiComposite, SWT.WRAP | SWT.RIGHT | SWT.VERTICAL);
-    PropsUi.setLook(perspectivesToolbar, Props.WIDGET_STYLE_TOOLBAR);
-    FormData fdToolBar = new FormData();
-    fdToolBar.left = new FormAttachment(0, 0);
-    fdToolBar.top = new FormAttachment(0, 0);
-    fdToolBar.bottom = new FormAttachment(100, 0);
-    perspectivesToolbar.setLayoutData(fdToolBar);
+    // Create custom sidebar composite instead of ToolBar for better control
+    perspectivesSidebar = new Composite(mainHopGuiComposite, SWT.NONE);
+    PropsUi.setLook(perspectivesSidebar);
+    perspectivesSidebar.setBackground(display.getSystemColor(SWT.COLOR_WIDGET_BACKGROUND));
+
+    // Use GridLayout for vertical stacking
+    org.eclipse.swt.layout.GridLayout sidebarLayout =
+        new org.eclipse.swt.layout.GridLayout(1, false);
+    sidebarLayout.marginWidth = 1;
+    sidebarLayout.marginHeight = 2;
+    sidebarLayout.verticalSpacing = 1; // Minimal spacing between buttons
+    perspectivesSidebar.setLayout(sidebarLayout);
+
+    FormData fdSidebar = new FormData();
+    fdSidebar.left = new FormAttachment(0, 0);
+    fdSidebar.top = new FormAttachment(0, 0);
+    fdSidebar.bottom = new FormAttachment(100, 0);
+    fdSidebar.width = 40; // Fixed width for sidebar (80% of 50px)
```

Comment:
Use zoom factor

fdSidebar.width = (int) (40 * PropsUi.getNativeZoomFactor()); // Fixed width for sidebar (80% of 50px)